vortex-expr/src/pruning/relation.rs

```diff
@@ -0,0 +1,66 @@
+use std::fmt::Display;
+use std::hash::Hash;
+
+use itertools::Itertools as _;
+use vortex_array::aliases::hash_map::HashMap;
+use vortex_array::aliases::hash_set::HashSet;
+
+#[derive(Debug, Clone)]
+pub struct Relation<K, V> {
+    map: HashMap<K, HashSet<V>>,
+}
+
+impl<K: Display, V: Display> Display for Relation<K, V> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            self.map.iter().format_with(",", |(k, v), fmt| {
+                fmt(&format_args!("{k}: {{{}}}", v.iter().format(",")))
+            })
+        )
+    }
+}
```

Comment:
I don't think we need this Display impl, if we kill the PruningPredicate display impl